compiler/llvm/clangUtil.cpp

```diff
@@ -2547,17 +2547,25 @@ void runClang(const char* just_parse_filename) {
     addFilteredArgs(clangCCArgs, args, parseOnly);
   }
 
-  // tell clang to use CUDA support
+  // tell clang to use CUDA/AMD support
   if (usingGpuLocaleModel()) {
     // Need to pass this flag so atomics header will compile
     clangOtherArgs.push_back("--std=c++11");
-    // Need to select CUDA mode in embedded clang to
+
+    // Need to select CUDA/AMD mode in embedded clang to
     // activate the GPU target
     clangOtherArgs.push_back("-x");
-    clangOtherArgs.push_back("cuda");
+    switch (getGpuCodegenType()) {
+      case GpuCodegenType::GPU_CG_CUDA:
+        clangOtherArgs.push_back("cuda");
+        break;
+      case GpuCodegenType::GPU_CG_AMD:
+        clangOtherArgs.push_back("hip");
+        break;
+    }
 
-    std::string cudaGPUArch = std::string("--cuda-gpu-arch=") + fCUDAArch;
-    clangOtherArgs.push_back(cudaGPUArch);
+    std::string gpuArch = std::string("--offload-arch=") + fCUDAArch;
```

Comment:
Is `fCUDAArch` really a CUDA specific thing or is this also used in HIP?

Also nit: Should be `fCudaArch` if we want to keep our acronym capitalization style consistent.

Reply:
It's used for CUDA as well. What should we do about the environment variables?

Reply:
I would propose renaming to CHPL_GPU_ARCH and fGpuArch.